src/jdk.sctp/unix/classes/sun/nio/ch/sctp/SctpMultiChannelImpl.java

```diff
@@ -373,6 +373,7 @@ public void kill() throws IOException {
             if (state == ChannelState.KILLED)
                 return;
             if (state == ChannelState.UNINITIALIZED) {
+                SctpNet.close(fdVal);
                 state = ChannelState.KILLED;
```

Comment:
Good catch. It seems that there was a missing state here, but not worth adding at this point.

Reply:
I also thought that it looks like there is a missing state.